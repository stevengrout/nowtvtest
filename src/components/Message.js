import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import './message.css';

const DATE_FORMAT = 'ddd, Do MMMM YYYY, h:mmA';

const formatTimestamp = (timestamp) => {
  const date = new Date(timestamp)
  return moment(date).format(DATE_FORMAT);
}

const MessageComponent = (props) => {
  return (
    <div className='message'>
      <div className='message-text'>
        <div className='tooltip'>{props.message}
          <div className='tooltiptext'>{props.email}</div>
        </div>
        <p>{formatTimestamp(props.time)}</p>
      </div>
      <div className='message-avatar'>
        <img src={props.avatar} alt={'missing avatar'}/>
      </div>
    </div>
  );
};

MessageComponent.propTypes = {
  message: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  avatar: PropTypes.string,
  time: PropTypes.string,
};

export default MessageComponent;
