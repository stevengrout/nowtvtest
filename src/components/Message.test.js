/* global describe it expect */

import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15.4';
import Enzyme, { shallow } from 'enzyme';
import Message from './Message';

Enzyme.configure({ adapter: new Adapter() });

describe('Message details', () => {
  it('should render all the properties', () => {
    const output = shallow((
      <Message message={'mockMessage'} avatar={'http://dummyimage.com/100x100.jpg/ff4444/ffffff'}
               time={'2016-04-07T10:20:42Z'} email={'test.me@testing.com'}/>
    ));
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(output.prop('class'));
  });
});
