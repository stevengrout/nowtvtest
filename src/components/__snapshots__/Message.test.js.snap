exports[`Message details should render all the properties 1`] = `
<div
  className="message">
  <div
    className="message-text">
    <div
      className="tooltip">
      mockMessage
      <div
        className="tooltiptext">
        test.me@testing.com
      </div>
    </div>
    <p>
      Thu, 7th April 2016, 11:20AM
    </p>
  </div>
  <div
    className="message-avatar">
    <img
      alt="missing avatar"
      src="http://dummyimage.com/100x100.jpg/ff4444/ffffff" />
  </div>
</div>
`;
